solr/core/src/java/org/apache/solr/search/SolrDocumentFetcher.java

```diff
@@ -609,7 +609,7 @@ private Object decodeDVField(
       case BINARY:
         BinaryDocValues bdv = e.getBinaryDocValues(localId, leafReader, readerOrd);
         if (bdv != null) {
-          return BytesRef.deepCopyOf(bdv.binaryValue());
+          return BytesRef.deepCopyOf(bdv.binaryValue()).bytes;
```

Comment:
Previously this code was returning `BytesRef` objects that were serialized up the stack with a simple `.toString` into `BytesRef@<hash-code>` string. You can revert this change and see it yourself in a test failure.

Now we return a `byte[]` which is correctly supported by SolrJ client and serialized as base64.

Note that there's no any additional performance overhead as we were already doing `deepCopy` into a new byte array and .bytes simply return a reference to a byte array.